### ISSUE TITLE
Docker image runs as root

### ISSUE THREAD

**Issue by ianlewis** [silverbulletmd/silverbullet#1170]
The current `Dockerfile` extends the `denoland/deno:debian-2.0.6` image which prepares a user `deno:deno` but this user is not set in the base image via a `USER` directive. This is left to child images as described in the [`denoland/deno` README](https://github.com/denoland/deno_docker/tree/main?tab=readme-ov-file#as-a-dockerfile). However, the SilverBullet image fails to set this as well.

After building the image:
```
$ docker run --entrypoint id 088e52e57b9f
uid=0(root) gid=0(root) groups=0(root)
```

SilverBullet should use the user provided by the base image and run as `deno:deno` by default.

**Comment by zefhemel** [comment]
Have a look at the docker section: https://silverbullet.md/Install/Configuration

It's possible to configure both the UID and GID of the user used in the container using PUID and PGUID environment variables.

**Comment by ianlewis** [comment]
Yeah, I saw that there is some logic in the docker endpoint to setuid/setgid before starting the app but I think I'd prefer that the endpoint be run as non-root rather than having the endpoint setuid/setgid in the container.

I'm also aware you can override the image's uid/gid via flags to Docker but some container platforms like Cloud Run don't allow for using this override and rely on the image uid/gid.

**Comment by ivangretsky** [comment]
@zefhemel I am experiencing this once again on SilverBullet 2.10.0 running with docker compose. The  `/space` root is owned by a non-root user. But all the new pages are created under root. The old ones were under non-root user.

Setting PUID=1000 and PGID=1000 explicitly doesn't help.

Could be a regression? Or a unintended consequence of multi-space?

**Comment by zefhemel** [comment]
It was a regression and has once again been fixed, sorry about that.
